src/query/models/tag.go

```diff
@@ -145,17 +146,47 @@ func (m Matchers) ToTags() (Tags, error) {
 	return Normalize(tags), nil
 }
 
-// ID returns a string representation of the tags
-func (t Tags) ID() string {
-	b := make([]byte, 0, len(t))
+// StringID returns a string representation of the tags
+func (t Tags) StringID() string {
+	var (
+		idLen      = t.IDLen()
+		strBuilder = strings.Builder{}
+	)
+
+	strBuilder.Grow(idLen)
+	for _, tag := range t {
+		strBuilder.WriteString(tag.Name)
+		strBuilder.WriteByte(eq)
+		strBuilder.WriteString(tag.Value)
+		strBuilder.WriteByte(sep)
+	}
+
+	return strBuilder.String()
+}
+
+// WriteBytesID writes out the ID representation
+// of the tags into the provided buffer.
+func (t Tags) WriteBytesID(b []byte) []byte {
 	for _, tag := range t {
 		b = append(b, tag.Name...)
 		b = append(b, eq)
 		b = append(b, tag.Value...)
 		b = append(b, sep)
 	}
 
-	return string(b)
+	return b
+}
+
+// IDLen returns the length of the ID that would be
+// generated from the tags.
+func (t Tags) IDLen() int {
+	idLen := 0
```

Comment:
nit: might be a little better to have `idLen := len(t) * 2`

Reply:
I think thats wrong though, because I want it to be the length of the generated ID, not twice the number of tags

Reply:
Yeah, sorry I mean initialize it to be twice the length of `t` to account for `eq` and `sep` rather than adding 2 each time through the loop